dietpi/dietpi-ddns

```diff
@@ -307,17 +324,23 @@ Menu_Provider()
 
 Menu_Domains()
 {
-	# Skip with FreeDNS and custom provider
-	[[ $PROVIDER == 'DuckDNS' || $PROVIDER == 'No-IP' || $PROVIDER == 'Dynu' || $PROVIDER == 'OVH' ]] || return 0
+	# Skip with these providers
+	[[ $PROVIDER == 'FreeDNS' ]] && return 0
```

Comment:
It is not beautiful, but let's keep the previous method, otherwise the menu is not skipped with custom providers:
```suggestion
	# Skip with FreeDNS and custom provider
	[[ $PROVIDER == 'DuckDNS' || $PROVIDER == 'No-IP' || $PROVIDER == 'Dynu' || $PROVIDER == 'YDNS' || $PROVIDER == 'OVH' ]] || return 0
```
When one finds time, we should create some clear arrays at the start of the script which contains the information for each provider whether it supports domain, username, password and/or token input, the API URL and such. So adding a new provider could be done with just adding array entries in one place in the code and the menus are correctly generated based on the arrays' content. Then we can also get rid of these growing conditions in the code.

Reply:
Fair enough! Agreed that it would be good to refactor this especially since misspelling an entry across the file can easily cause a tricky bug.

Reply:
Applied the change to my latest commit, but added YDNS to the list too :)